packages/devtools_app/lib/src/screens/inspector/inspector_screen.dart

```diff
@@ -174,6 +176,13 @@ class InspectorScreenBodyState extends State<InspectorScreenBody>
         ),
       ],
     );
+    preferences.inspector.customPubRootDirectories.addListener(() {
+      _refreshInspector();
+    });
+    serviceManager.isolateManager.isolates.addListener(() {
+      _refreshInspector();
+      preferences.inspector.loadCustomPubRootDirectories();
+    });
```

Comment:
In general, we should never be doing anything with listeners inside of `build`. This code here would result in a new listener being called each time we build the inspector (which would be quite often). 

Any code like this should be in `initState` and should be added with `addAutoDisposeListener` - other wise the listener will never be disposed and this will lead to memory leaks.

Reply:
great catch on this!